[CS2113-T16-1] FinanceIt

kaiwen98 opened the pull request:
FinanceIt is an all-in-one desktop application that handles the finance tracking needs of university students who are comfortable with a CLI interface.
	It includes 5 different finance tools, all of which take in typed commands from the users to execute their respective functions.

daniellimws reviewed (commented):
![review1](https://user-images.githubusercontent.com/8130120/97390387-b1088580-1917-11eb-8cc3-5ffe3e9973dc.png)

Referring to *Overall Architecture.png*, is the relationship with `ParamChecker` supposed to be a dependency instead of association, since it is a `Singleton`?

daniellimws commented:
Similar to my comment on *Overall Architecture.png*, in *Handler_Commands.png*, are the `Command` classes supposed to be dependencies instead of association?

daniellimws commented:
![review2](https://user-images.githubusercontent.com/8130120/97391557-6fc5a500-191a-11eb-8b0a-09d1541f457a.png)

For the sequence diagrams, iirc the invocation needs to come from either an activation bar or from an actor. Like this taken from the textbook.

![image](https://user-images.githubusercontent.com/8130120/97391676-badfb800-191a-11eb-9277-6c7e6f59120e.png)

 In the sequence diagrams in this DG, the first invocation comes from the lifeline, as I annotated in the image above. Not sure whether this is correct.

daniellimws commented on docs/DeveloperGuide.md:
````diff
@@ -1,18 +1,541 @@
+the respective month to ensure that the user does not overspent his budget. 
+When user enter ```expense 2000 for 08```, the command will be sent to InputParser and parse it into String[].
+With the String[], it will be sent to a class called ```Goal```, and it will store the individual information.
+Afterwards, it will be added to a ArrayList in a class called ```TotalGoalList```. 
````
Should String[] and InputParser be in a code block (as in `String[]` and `InputParser)?

daniellimws commented on docs/DeveloperGuide.md:
```diff
@@ -1,18 +1,541 @@
+Secondly, the input command will be sent to InputParser to parse.
+Thirdly, the parsed information will be sent to class `Goal` to store the individual information
+Next, it will be added to a ArrayList in class `TotalGoalList`.
+Lastly, the goal status will be displayed to the user.  
```
Should InputParser and ArrayList be in code blocks? (`ArrayList` and `InputParser`)

daniellimws commented on docs/DeveloperGuide.md:
````diff
@@ -1,18 +1,541 @@
+own class declarations.
+
+##### Architecture
+* The initialisation of ```Ledger``` and ```Entry``` instances can be
+performed with reference to input parameters supplied from the user input.
+* For ledger creation operations, the input from the user is parsed and passed into an initialized ledger instance
+to handle. That is, the handling of input parameters is abstracted out from the tracker classes. 
+<br> The handle operation will set the various attributes within the ledger in accordance to specifications inferred
+from the user input. 
+<br> If the ledger is successfully specified in full, it will be added to a ```ledgerList``` instance within the handler 
+class ```ManualTracker```.
+* For ledger deletion/open, a ledger will need to be selected from the ledger list maintained by the handler class.
+<br>Hence, the input from the user is parsed and passed into a command instance to handle. If the input
+is valid, the ledger list instance will assign a reference to the ledger selected to a public ```currLedger``` 
+attribute. 
+<br>After which, an operation of edit/open would be performed upon the ledger referenced from 
+```currLedger``` in ```ledgerList```.
````
Maybe this is still work in progress. I feel that this would be easier to understand with relevant diagrams.

acyang97 replied:
Same here, I think something like a flow chart would be good in aiding understanding

daniellimws commented on docs/DeveloperGuide.md:
````diff
@@ -1,18 +1,541 @@
+```
+    <command> <param type> <parameter> <param type> <parameter> ...
+```
+* The ```command``` string determines the current state of the Finite State Machine, and
````
For this reference to a Finite State Machine, I think it would be good to include some diagrams to help the reader (me) visualize it.

daniellimws commented on docs/DeveloperGuide.md:
````diff
@@ -1,18 +1,541 @@
+| ```Data``` |Refers to ```Ledger``` instances, stores relevant data of the day's transactions.
+| ```Data List``` |Refers to ```LedgerList``` instances, maintains Ledger instances within the program. 
+| ```Commands``` |Processes information from ```CommandPacket``` and executes the appropriate process from recognised params.
+| ```Logic``` |Outlines the abstract behavior of commands, as well as handle verification of params with appropriate error handling.
````
I feel that it would be good to have one sentence to give an overview of this section.

daniellimws commented on docs/DeveloperGuide.md:
````diff
@@ -1,18 +1,541 @@
+|```createEntryCommand```| Omitted for brevity.
+|```editEntryCommand```| Omitted for brevity.
+|```ParamChecker```| Class contains a collection of methods that verify the correctness of the ```param``` supplied. <br><br> For instance, ```ParamChecker.checkAndReturnIndex``` checks if the index provided is out of bounds relative to the specified list, and throws the relevant exception if the input index is invalid. 
+|```ParamHandler```| Abstract class that outlines the general param handling behavior of ```commands``` instances and other classes that need to handle ```params``` in its operation.  
````
I feel that it would be good to have one sentence to give an overview of this section.

daniellimws commented on docs/DeveloperGuide.md:
````diff
@@ -1,18 +1,541 @@
+|```createEntryCommand```| Omitted for brevity.
+|```editEntryCommand```| Omitted for brevity.
+|```ManualTracker```| Implements Manual Tracker. Contains handler methods that implements a particular operation capable by the Manual Tracker. <br><br> These methods use the above ```command``` instances for param handling operations from user input.
+|```EntryTracker```| Omitted for brevity.
````
I feel that it would be good to have one sentence to give an overview of this section.

daniellimws commented on docs/DeveloperGuide.md:
````diff
@@ -1,18 +1,541 @@
+|```DateTimeItem```| Abstract class that extends ```Item``` class; instances will have ```LocalDate``` or ```LocalTime``` attributes and corresponding helper methods.
+|```Item```| Abstract class to define behavior of entities that need are stored in ```ItemList``` instances.
+
+#### Functions with Sequence Diagrams
````
Perhaps something like "Details of implementation" would better represent this section as compared to "Functions with Sequence Diagrams"

daniellimws commented on docs/DeveloperGuide.md:
````diff
@@ -1,18 +1,541 @@
+* ```/p``` - Number of Months (Mandatory)
+* ```/d``` - Monthly Deposit (Optional)
+
+The following class diagram shows how the Yearly/Monthly Compound Interest Calculator feature works:
````
I feel that a jump from parameters to class diagram is quite abrupt. Perhaps it would be good to add another h3 header.

daniellimws reviewed (commented):
The developer guide looks quite comprehensive! It seems that every module in the codebase is well-documented with a lot of details, which helps me to know a lot about the system before even reading the code. 👍 

I feel that it was sometimes hard to navigate because most of the headings are around the same level. Perhaps some whitespace or a horizontal line could be added to separate each section more clearly.

I also feel that for most sections, the level of the headings don't accurately represent the level of hierachy of the content, making the flow feel quite weird. For example under `### FinanceTools`, every header is a level 3 header (`###`), i.e. `### Simple Interest Calculator` and `### Parameters` are also level 3 headers. But I got the impression that parameters should be one level below `### SimpleInterestCalculator` (i.e. `#### Parameters` instead). So I am quite confused with regards to this.

I also face this problem with most of the other sections in the guide.

acyang97 commented on docs/DeveloperGuide.md:
````diff
@@ -1,18 +1,616 @@
+        * Value:  ```param```
+
+##### InputParser
+* A helper class. Parses the input string and returns a corresponding ```commandPacket```.
````
I'm actually quite confused as to whether commandPacket is suppose to be a property of a class or a class itself, seems like it is a class after reading at the ParamHandler component, but from this line it seems like  a property of some other class.

acyang97 commented on docs/DeveloperGuide.md:
````diff
@@ -1,18 +1,616 @@
+
+|Class| Function |
+|--------|----------|
+|```retrieveLedgerCommand```| Process ```paramTypes```-```param``` pairs from the ```CommandPacket``` instance to identify specified ```Ledger``` instance, then retrieves the instance from the existing ```LedgerList```.
+|```createLedgerCommand```| Process ```paramTypes```-```param``` pairs from the ```CommandPacket``` instance to identify specified ```Ledger``` instance to be created, then creates the instance and append to existing ```LedgerList```.
+|```retrieveEntryCommand```| Omitted and left as exercise for reader. : ^ )
````
Classes should be written woth first letter of each internal word in upper case

acyang97 commented on docs/DeveloperGuide.md:
```diff
@@ -1,18 +1,616 @@
+
+##### Handler and Command
+
+![](uml_images/manualTracker/images/Handler_Commands.png)
```
Is retriveEntryCommand, editEntryCOmmand, createEntryCommand properties of EntryTracker? Doesn't seem to be very clear based on the diagram as the table below doesn't really states how any of these EntryCommands are being used/ returned, etc. Also, is State a class or an Enumaration or? Would be better to make it clearer earlier

acyang97 commented on docs/DeveloperGuide.md:
```diff
@@ -1,18 +1,616 @@
+
+##### Handler and Parser
+
+![](uml_images/manualTracker/images/Handler_Parser.png)
```
Might want to indicate the multiplicity clearer to make everything nicer

acyang97 commented on docs/DeveloperGuide.md:
```diff
@@ -1,18 +1,616 @@
+
+##### Handler and Data
+
+![](uml_images/manualTracker/images/Handler_Data.png)
```
I think a better option might be to put the aggregation symbol from LedgerList to Ledger and EntryList to Entry (by putting a shaded diamond at LedgerList in the arrow pointing from LedgerList to Ledger).

HengFuYuen replied:
Perhaps you can consider making the association between EntryList and Ledger more specific. Based on the methods and attributes, it seems that Ledger contains an EntryList only but the multiplicity both ways is 1, does this mean that EntryList also stores a Ledger?

![image](https://user-images.githubusercontent.com/60005925/97536882-cbfef680-19f8-11eb-9298-d2489064797c.png)

HengFuYuen commented on docs/DeveloperGuide.md:
```diff
@@ -1,18 +1,616 @@
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Summary
```
Perhaps consider adding in some architecture diagrams or class diagrams for the Summary and Architecture sections so it is easier for the reader to visualise the information.

HengFuYuen commented on docs/DeveloperGuide.md:
````diff
@@ -1,18 +1,616 @@
+##### Input Conventions
+* The user input is composed of the following format:
+```
+    <command> <param type> <parameter> <param type> <parameter> ...
````
Perhaps you can consider adding specific examples to aid reader understanding.

HengFuYuen commented on docs/DeveloperGuide.md:
```diff
@@ -1,18 +1,616 @@
+
+##### Command and Logic
+
+![](uml_images/manualTracker/images/Commands_Logic_edited.png)
```
Regarding this class diagram,  based on the dashed line from the various commands to the ParamChecker, I am assuming that the command classes have dependencies with the ParamChecker. In this case, is showing multiplicities for dependencies correct? Perhaps you might want to consider removing it.  You can also consider this change for the class diagram under the handler and parser section. Perhaps you may also want to consider capitalising the first letter of the command classes. Also, I am not too sure if the current notation of singleton is correct or not.

![image](https://user-images.githubusercontent.com/60005925/97535953-63634a00-19f7-11eb-9cf6-c01d556ec5bf.png)

HengFuYuen commented on docs/DeveloperGuide.md:
```diff
@@ -1,18 +1,616 @@
+
+#### Functions with Sequence Diagrams
+
+##### Creation of Ledger
```
Perhaps you can consider having object diagrams for each step and you can also consider this for the other functions as well.

HengFuYuen commented on docs/DeveloperGuide.md:
```diff
@@ -1,18 +1,616 @@
+
+
+
+#### Feature 2: FinanceTools
```
Perhaps this might be feature 3 instead?

qkbrandon replied:
Noted, thanks for correcting!

HengFuYuen commented on docs/DeveloperGuide.md:
```diff
@@ -1,18 +1,616 @@
+
+The following class diagram shows how the Simple Interest Calculator feature works:
+<br />
+![ClassDiagram](uml_images/financetools/SimpleInterest/SimpleInterestClassDiagram.png)
```
Perhaps you would like to take a look at this issue in the class forum about icons in PUML and consider the changes for the diagrams drawn in the following few sections:
https://github.com/nus-cs2113-AY2021S1/forum/issues/112

qkbrandon replied:
Thanks a lot for this! Missed it inititally

HengFuYuen commented on docs/DeveloperGuide.md:
```diff
@@ -1,18 +1,616 @@
+<br />
+The following sequence diagram shows how the Simple Interest Calculator feature works:
+<br />
+![SequenceDiagram2](uml_images/financetools/SimpleInterest/SimpleInterestSequenceDiagram(2).png)
```
Perhaps you can consider retaining only the object box at the top. You can also consider this change for the diagram above this.

![image](https://user-images.githubusercontent.com/60005925/97537971-965b0d00-19fa-11eb-9f5c-8c56f499c638.png)

qkbrandon replied:
Got it, thanks!

HengFuYuen commented on docs/DeveloperGuide.md:
````diff
@@ -1,18 +1,616 @@
+```FinanceTools.main()```.
+<br />
+
+__Parameters (Yearly Compound Interest Calculator)__
````
Perhaps you can consider combining the two similar parts in this section to reduce duplicates.

qkbrandon replied:
Thank you!

HengFuYuen commented on docs/DeveloperGuide.md:
```diff
@@ -1,18 +1,616 @@
+The commands are stored before the params are handled and implementation is executed. The results from calculation
+is stored when the implementation has finished executed.
+
+#### Feature 3: Goal Tracker
```
Perhaps this is feature four maybe?

HengFuYuen commented on docs/DeveloperGuide.md:
```diff
@@ -1,18 +1,616 @@
+* setExpenseGoal: expense 5000 for 08
+* setIncomeGoal: income 5000 for 08
+ 
+##### How it works?
```
Perhaps consider using object diagrams.

HengFuYuen commented on docs/DeveloperGuide.md:
```diff
@@ -1,18 +1,616 @@
+This sequence diagram will show the flow of setting of expense goal:
+![ExpenseSequenceDiagram](uml_images/goaltracker/SetExpenseGoalSequenceDiagram.png)
+
+#### Feature 4: Save Manager
```
Perhaps this is feature five?

HengFuYuen replied:
Perhaps consider including a higher level diagram.

HengFuYuen reviewed (commented):
A lot of effort has been put into the developer guide. Perhaps some changes might make it better but good job!

Feudalord committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T16-1/tp

Feudalord committed:
Merge branch 'master' of https://github.com/Feudalord/tp

Artemis-Hunt committed:
Add validation for day of month inputted.
Add warning for day of month between 29 and 31, since some month(s) do not have these day(s)
Fix formatting and reword error message for clarity.
Add excluded months into list for RecurringTracker

Artemis-Hunt committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T16-1/tp

Artemis-Hunt committed:
Fix checkstyle errors.

Artemis-Hunt committed:
Merge pull request #248 from Artemis-Hunt/master

Bug fixes for RecurringTracker

Artemis-Hunt committed:
Update manual testing section of dev guide.
Fixed UI bugs relating to table printing

Artemis-Hunt committed:
Merge pull request #249 from Artemis-Hunt/master

Update manual testing section of dev guide.

Commit:
Handle negative and alphabet inputs in FinanceTools

qkbrandon committed:
Merge pull request #250 from bqxy/finance-tools

Handle negative and alphabet inputs in FinanceTools

Commit:
Add Table of Content in UserGuide.md

Commit:
Edit Table of Contents in UserGuide.md

Commit:
Edit Table of Contents in UserGuide.md

qkbrandon committed:
Merge pull request #251 from bqxy/finance-tools

Update Table of Contents in UserGuide.md

Commit:
Edit UserGuide.md

Commit:
Edit UserGuide.md

Commit:
Edit UserGuide.md

Commit:
Add Table of Contents for DeveloperGuide.md

Commit:
Edit DeveloperGuide.md

Commit:
Edit DeveloperGuide.md

Commit:
Updated DG to fit new standards

Commit:
Edit DeveloperGuide.md

Commit:
Resize diagrams

Commit:
Edit DeveloperGuide.md

qkbrandon committed:
Merge pull request #252 from bqxy/finance-tools

Add Table of Contents for DeveloperGuide.md

Commit:
Edit AccountStorage.java

Handle empty account names and notes
Remove scientific notation display

Commit:
Edit UserGuide.md

Added more explanation for cashback feature in FinanceTools

qkbrandon committed:
Merge pull request #253 from bqxy/finance-tools

Edit AccountStorage.java in FinanceTools

Commit:
Update Screenshots in UserGuide.md for FinanceTools

qkbrandon committed:
Merge pull request #254 from bqxy/finance-tools

Update Screenshots in UserGuide.md for FinanceTools

Commit:
Update DeveloperGuide.md

Include Manual Testing for FinanceTools
Add more parser explanation in FinanceTools

Commit:
Resolved parser bugs and made ManualTracker more defensive to corner cases

Commit:
Edit DeveloperGuide.md

qkbrandon committed:
Merge pull request #255 from bqxy/finance-tools

Update DeveloperGuide.md

Commit:
Update DeveloperGuide.md

Minor aesthetic touch up

qkbrandon committed:
Merge pull request #256 from bqxy/finance-tools

Update DeveloperGuide.md

Commit:
Resolved more parser bugs and made ManualTracker much more defensive to corner cases

kaiwen98 committed:
Merge branch 'master' into master

kaiwen98 committed:
Merge pull request #257 from kaiwen98/master

Resolved all issues pertaining ManualTracker, EntryTracker and Parser classes from PE dry run

Commit:
Update DeveloperGuide.md

Updated sequence diagrams for features in FinanceTools

qkbrandon committed:
Merge pull request #258 from bqxy/finance-tools

Update DeveloperGuide.md

Commit:
remove print msgs

Commit:
Merge branch 'master' of https://github.com/AY2021S1-CS2113T-T16-1/tp

kaiwen98 committed:
Merge pull request #259 from kaiwen98/master

removed print messages

Commit:
Update UserGuide.md

RepoSense compatibility

Commit:
Update DeveloperGuide.md

RepoSense compatibility

Commit:
RepoSense compatibility for FinanceTools

Commit:
Edit to pass checkstyleMain

Commit:
Update bqxy.md

dixoncwc committed:
Did some slap for Goal Tracker. Add in future feature in UG and DG

qkbrandon committed:
Merge pull request #260 from bqxy/finance-tools

RepoSense Compatibility

dixoncwc committed:
Resolved CheckStyleMain

dixoncwc committed:
Merge pull request #261 from dixoncwc/GoalsTracker

Did some slap for Goals tracker. Add in future feature in UG and DG

Feudalord committed:
Add files via upload

Feudalord committed:
Add files via upload

Feudalord committed:
Update DeveloperGuide.md

dixoncwc committed:
Added in some details to DG

Feudalord committed:
Update DeveloperGuide.md

Feudalord committed:
Update DeveloperGuide.md

dixoncwc committed:
Merge pull request #262 from dixoncwc/GoalsTracker

Added in some details to DG

Feudalord committed:
Update DeveloperGuide.md

kaiwen98 committed:
Update UserGuide.md

kaiwen98 committed:
Update UserGuide.md

Artemis-Hunt committed:
Add more JUnit tests for ParamChecker, RecurringTracker and ReminderGenerator.
Update JavaDoc comments

Artemis-Hunt committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T16-1/tp

kaiwen98 committed:
Update UserGuide.md

kaiwen98 committed:
Update UserGuide.md

Commit:
Updated DG

Commit:
Updated DG

Artemis-Hunt committed:
Improve UML diagrams for Logic, data, handler

Artemis-Hunt committed:
Merge pull request #263 from Artemis-Hunt/master

Improve UML diagrams. Code cleanup

Feudalord committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T16-1/tp

# Conflicts:
#	data/backup/names.txt

Commit:
Updated DG

Commit:
Added more comments to code

Feudalord committed:
AddedComments

Feudalord committed:
checkstyle

Feudalord committed:
Merge pull request #265 from Feudalord/master

Storage Component Documentation

Commit:
Fixed merge conflict

Commit:
UG updates

kaiwen98 committed:
Update DeveloperGuide.md

kaiwen98 committed:
Update DeveloperGuide.md

kaiwen98 committed:
Update DeveloperGuide.md

Commit:
Updated DG

kaiwen98 committed:
Update DeveloperGuide.md

kaiwen98 committed:
Update DeveloperGuide.md

kaiwen98 committed:
Update DeveloperGuide.md

kaiwen98 committed:
Update DeveloperGuide.md

kaiwen98 committed:
Update DeveloperGuide.md

Commit:
Update UserGuide.md & RepoSense Compatibility

qkbrandon committed:
Merge pull request #266 from bqxy/finance-tools

Update UserGuide.md & RepoSense Compatibility

Artemis-Hunt committed:
Fix checkstyle errors.
Updated DG and UML diagrams

Artemis-Hunt committed:
Fix checkstyle errors.
Updated DG and UML diagrams

Artemis-Hunt committed:
Merge branch 'master' into master

Artemis-Hunt committed:
Merge pull request #267 from Artemis-Hunt/master

Updated DG. Fix checkstyle

Commit:
Updated Developer's guide

Commit:
Merge branch 'master' of https://github.com/kaiwen98/tp

kaiwen98 committed:
Update DeveloperGuide.md

kaiwen98 committed:
Update DeveloperGuide.md

kaiwen98 committed:
Update DeveloperGuide.md

kaiwen98 committed:
Update DeveloperGuide.md

Artemis-Hunt committed:
Update DG.
Add UML diagrams for RecurringTracker

Artemis-Hunt committed:
Merge branch 'master' of https://github.com/Artemis-Hunt/tp

Artemis-Hunt committed:
Merge pull request #268 from Artemis-Hunt/master

Update DG. Add UML

Commit:
Updated DG

Commit:
Updated DG

Commit:
Updated DG

Commit:
Contributions to DG

Commit:
UG contributions

Artemis-Hunt committed:
Update UG and DG for RecurringTracker.
Add screenshots

kaiwen98 committed:
Update kaiwen98.md

kaiwen98 committed:
Merge pull request #264 from kaiwen98/master

Developer's Guide updates

kaiwen98 committed:
Update kaiwen98.md

Commit:
Contributions to ug

Commit:
Updated DG

Commit:
Updated DG

kaiwen98 committed:
Merge branch 'master' into master

kaiwen98 committed:
Update UserGuide.md

kaiwen98 committed:
Merge pull request #269 from kaiwen98/master

Updated DG

kaiwen98 committed:
Create README.md

kaiwen98 committed:
Update README.md

kaiwen98 committed:
Create kaiwen98.md

kaiwen98 committed:
Create kaiwen98

kaiwen98 committed:
Update README.md

kaiwen98 committed:
Delete README.md

kaiwen98 committed:
Set theme jekyll-theme-slate

kaiwen98 committed:
Create kaiwen98.md

kaiwen98 committed:
Delete kaiwen98.md

kaiwen98 committed:
Rename kaiwen98 to kaiwen98.md

kaiwen98 committed:
Set theme jekyll-theme-slate

kaiwen98 committed:
Update DeveloperGuide.md

Artemis-Hunt committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T16-1/tp

# Conflicts:
#	docs/DeveloperGuide.md

Artemis-Hunt committed:
Update DG/UG

Artemis-Hunt committed:
Merge pull request #270 from Artemis-Hunt/master

Update DG/UG

Feudalord committed:
Implemented dynamic save. And loading by line

Feudalord committed:
Fixed disruptive Exceptions for testing and make exceptions more useful

Feudalord committed:
Improved Exception

Feudalord committed:
found the culprit for mess up test case

Feudalord committed:
Merge pull request #271 from Feudalord/master

Implemented dynamic save and loading by line

Commit:
Implemented tests for redundant edit commands

Commit:
Made code more defensive

Commit:
Made code more defensible

dixoncwc committed:
no message

dixoncwc committed:
Add in details for AboutUs

Commit:
finalised code

dixoncwc committed:
Merge pull request #272 from dixoncwc/GoalsTracker

Add in details for AboutUs

dixoncwc committed:
Edited some details in AboutUs

dixoncwc committed:
Edited AboutUs

Commit:
finalised code

dixoncwc committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113T-T16-1/tp into GoalsTracker

kaiwen98 committed:
Merge pull request #273 from kaiwen98/final

Final

dixoncwc committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113T-T16-1/tp into GoalsTracker

dixoncwc committed:
Merge pull request #274 from dixoncwc/GoalsTracker

Edited some details in AboutUs

dixoncwc committed:
Edited my PPP

dixoncwc committed:
Merge pull request #275 from dixoncwc/GoalsTracker

Edited my PPP

Commit:
updated dg image

kaiwen98 committed:
Merge pull request #276 from kaiwen98/master

Updated DG image

kaiwen98 committed:
Update UserGuide.md

kaiwen98 committed:
Update DeveloperGuide.md

kaiwen98 committed:
Update kaiwen98.md

kaiwen98 committed:
Update kaiwen98.md

Feudalord committed:
Update and rename johndoe.md to sly.md

kaiwen98 committed:
Update kaiwen98.md

kaiwen98 committed:
Set theme jekyll-theme-slate

kaiwen98 committed:
Set theme jekyll-theme-cayman

kaiwen98 committed:
Update README.md

kaiwen98 committed:
Update AboutUs.md

kaiwen98 committed:
Update kaiwen98.md

kaiwen98 committed:
Update kaiwen98.md

kaiwen98 committed:
Update AboutUs.md

kaiwen98 committed:
Update AboutUs.md

Feudalord committed:
Update DeveloperGuide.md

Commit:
updated ug

Commit:
Merge branch 'master' of https://github.com/AY2021S1-CS2113T-T16-1/tp into final

Commit:
updated PPP

Commit:
Updated ppp

kaiwen98 committed:
Merge pull request #277 from kaiwen98/final

Final

Feudalord committed:
Merge pull request #278 from Feudalord/master

Update DeveloperGuide.md

Feudalord committed:
Update sly.md

Artemis-Hunt committed:
Update and rename Artemis-Hunt.md to artemis-hunt.md

Artemis-Hunt committed:
Fix checkstyle

Artemis-Hunt committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T16-1/tp

Artemis-Hunt committed:
Merge pull request #279 from Artemis-Hunt/master

Fix checkstyle. Update PPP

Commit:
updated dg size

Commit:
Adjusted img

Commit:
Adjusted img

Artemis-Hunt committed:
Update AboutUs.md

Artemis-Hunt committed:
Add files via upload

Artemis-Hunt committed:
Update AboutUs.md

Artemis-Hunt committed:
Update AboutUs.md

Artemis-Hunt committed:
Update AboutUs.md

Artemis-Hunt committed:
Update AboutUs.md

Artemis-Hunt committed:
Update AboutUs.md

Commit:
Updated img

kaiwen98 committed:
Merge pull request #280 from kaiwen98/master

Adjusted DG images size

Feudalord committed:
Update sly.md

Feudalord committed:
Update sly.md

Feudalord committed:
Update sly.md

kaiwen98 committed:
Set theme jekyll-theme-slate

kaiwen98 committed:
Set theme jekyll-theme-slate

Feudalord committed:
Update sly.md

Artemis-Hunt committed:
Update artemis-hunt.md

Feudalord committed:
Update sly.md

Feudalord committed:
Update sly.md

Feudalord committed:
Update AboutUs.md

Feudalord committed:
Update AboutUs.md

Feudalord committed:
Update sly.md

Feudalord committed:
Update sly.md

Commit:
Merge branch 'master' of https://github.com/AY2021S1-CS2113T-T16-1/tp into final

Commit:
png

Commit:
png

kaiwen98 committed:
Merge pull request #281 from kaiwen98/final

Updated images on entry commands

kaiwen98 committed:
Update DeveloperGuide.md

kaiwen98 committed:
Update DeveloperGuide.md

kaiwen98 committed:
Update DeveloperGuide.md

kaiwen98 committed:
Update DeveloperGuide.md

kaiwen98 committed:
Update DeveloperGuide.md

kaiwen98 committed:
Update DeveloperGuide.md

kaiwen98 committed:
Update DeveloperGuide.md

kaiwen98 committed:
Update DeveloperGuide.md

Commit:
Update DeveloperGuide.md

Add User Stories

Artemis-Hunt committed:
FIx bug in RecurringTrackerTest

Artemis-Hunt committed:
Fix checkstyle

qkbrandon committed:
Merge pull request #283 from bqxy/finance-tools

Update DeveloperGuide.md

Artemis-Hunt committed:
Merge pull request #282 from Artemis-Hunt/master

FIx bug in RecurringTrackerTest

Commit:
Update UserGuide.md

Add page breaks

Commit:
Update UserGuide.md

Add page breaks

qkbrandon committed:
Merge pull request #284 from bqxy/finance-tools

Add page breaks for UserGuide.md

kaiwen98 committed:
Update UserGuide.md

kaiwen98 committed:
Update UserGuide.md

Commit:
Update DeveloperGuide.md

Add page breaks

qkbrandon committed:
Merge pull request #285 from bqxy/finance-tools

Add page breaks for DeveloperGuide.md

Feudalord committed:
Image Resize

Feudalord committed:
Merge branch 'master' of https://github.com/Feudalord/tp

Feudalord committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T16-1/tp

Feudalord committed:
Merge pull request #286 from Feudalord/master

Image Resize

Commit:
Resized images

kaiwen98 committed:
Merge pull request #287 from kaiwen98/final

Resized images

Artemis-Hunt committed:
Update DG

Artemis-Hunt committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T16-1/tp

Artemis-Hunt committed:
Update widths for images in DG/UG

Artemis-Hunt committed:
Fix bug relating to RecurringTrackerTest

Artemis-Hunt committed:
Merge pull request #288 from Artemis-Hunt/master

Update DG

kaiwen98 committed:
Update DeveloperGuide.md

kaiwen98 committed:
Update DeveloperGuide.md

kaiwen98 committed:
Update DeveloperGuide.md

kaiwen98 committed:
Update DeveloperGuide.md

kaiwen98 committed:
Update DeveloperGuide.md

kaiwen98 committed:
Update DeveloperGuide.md

Feudalord committed:
Saver Manager: Users can no longer enter special Characters

Feudalord committed:
Merge pull request #289 from Feudalord/master

Saver Manager: Users can no longer enter special Characters
